### PR TITLE
dl/populate countries

### PR DESCRIPTION
**Pull request by darrikonn**
# Populate select box with countries

## Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/shared/data/src/lib/countries.ts [hunk 1]
```diff
@@ -0,0 +1,998 @@
...
+  },
+  {
+    value: 'KE',
+    label: 'Kenya',
+  },
+  {
+    value: 'KI',
+    label: 'Kiribati',
+  },
+  {
```
Comment (arnifreyrs) [inline libs/shared/data/src/lib/countries.ts:470]: where is Kosovo? :/

File: libs/application/core/src/types/Fields.ts [hunk 2]
```diff
@@ -96,6 +96,7 @@ export interface SelectField extends BaseField {
   readonly type: FieldTypes.SELECT
   component: FieldComponents.SELECT
   options: MaybeWithApplication<Option[]>
+  defaultValue?: Option
```
Comment (arnifreyrs) [inline libs/application/core/src/types/Fields.ts:99]: nice, I like this. We will definitely need to add `defaultValue` to many other fields later on, maybe even `BaseField`

**Review by arnifreyrs — DISMISSED** [review]
great stuff

File: libs/shared/data/src/lib/countries.ts (hunk 1 above)
Comment (darrikonn) [reply libs/shared/data/src/lib/countries.ts:470]: Hahah, you just went through the whole country list :D ... good spot!

Comment (darrikonn) [reply libs/shared/data/src/lib/countries.ts:470]: Soooo.. which native part should I pick:
```
Republika e Kosovës  (Albanian)
Република Косово  (Serbian)
```

Comment (arnifreyrs) [reply libs/shared/data/src/lib/countries.ts:470]: Woah, tricky. Albanian is the most spoken language, so let's go for that https://en.wikipedia.org/wiki/Demographics_of_Kosovo#Languages

**Review by sindrig — COMMENTED** [review]
Shared data module sounds kinda scary TBH. Right now and for countries it's probably okay (not everyone relies on it and it needs some major event to change). But the name sounds ambiguous enough that someone might add all the shops in Kringlan at one point, and when one of the shops goes bankrupt we need to rebuild everything, even the modules that only rely on the country list and not the shops.

Just my 2c, maybe you want to think about renaming the module to something more concrete like `countries-list-data`.

**Comment by darrikonn** [comment]
> Shared data module sounds kinda scary TBH. Right now and for countries it's probably okay (not everyone relies on it and it needs some major event to change). But the name sounds ambiguous enough that someone might add all the shops in Kringlan at one point, and when one of the shops goes bankrupt we need to rebuild everything, even the modules that only rely on the country list and not the shops.
> 
> Just my 2c, maybe you want to think about renaming the module to something more concrete like `countries-list-data`.

Yeah, wasn't quite pleased with the name. But then again, I wasn't pleased with the name `shared` either since it's super broad. What is shared?

I agree, renaming the module is better!

**Comment by darrikonn** [comment]
Change in design, won't need it.

Agree with you @sindrig, the name was baaaad and too broad

**Pull request closed without merging**